Remove `svgDocument` interface from `HTMLFrameElement` to align with other browsers

Ahmad-S792 opened the pull request:
#### 11cb9ea590ddcb3b34e8631522a5fbd35fbd6e0c
<pre>
Remove `svgDocument` interface from `HTMLFrameElement` to align with other browsers

<a href="https://bugs.webkit.org/show_bug.cgi?id=278069">https://bugs.webkit.org/show_bug.cgi?id=278069</a>
<a href="https://rdar.apple.com/133800070">rdar://133800070</a>

Reviewed by Chris Dumez and Ryosuke Niwa.

This patch aligns WebKit with Web Specification [1]&quot;

[1] <a href="https://html.spec.whatwg.org/multipage/embedded-content-other.html#svg-0">https://html.spec.whatwg.org/multipage/embedded-content-other.html#svg-0</a>

As per web specification, `getSVGDocument` should only be exposed on
&apos;iframe&apos;, &apos;embed&apos; and &apos;object&apos;. This patch removes it from &apos;frame&apos; element.

Additionally, we made our implementation to return null in 259905@main but
never synced our IDL interface for &apos;object&apos; and &apos;embed&apos;, so this does and match
it with &apos;iframe&apos;.

* Source/WebCore/html/HTMLEmbedElement.idl:
* Source/WebCore/html/HTMLFrameElement.idl:
* Source/WebCore/html/HTMLObjectElement.idl:
* LayoutTests/svg/custom/frame-getSVGDocument-expected.txt -&gt; Renamed to interface-getSVGDocument-expected.txt:
* LayoutTests/svg/custom/frame-getSVGDocument.html -&gt; Renamed to interface-getSVGDocument.html: Now updated to &apos;testharness&apos; with all cases
* LayoutTests/platform/mac/svg/custom/frame-getSVGDocument-expected.png: Remove pixel output

Canonical link: <a href="https://commits.webkit.org/291498@main">https://commits.webkit.org/291498@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d7100491621202c5573b6cb92f6f4774c85d3d6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/93003 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/12554 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/2248 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/98002 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/43529 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/95053 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/12835 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/21007 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71106 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28523 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/96005 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/9670 "Found 1 new test failure: fast/forms/ios/focus-input-in-fixed.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84153 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51434 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9363 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/1788 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/42842 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79655 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/1764 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/100027 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/20055 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/14700 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80131 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/20307 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80051 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79432 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19741 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23979 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1282 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/13109 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/20039 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/25215 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/19726 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/23186 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/21467 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/html/HTMLFrameElement.idl:
```diff
@@ -34,6 +34,4 @@
     [CEReactions=NotNeeded, Reflect] attribute [LegacyNullToEmptyString] DOMString marginHeight;
     [CEReactions=NotNeeded, Reflect] attribute [LegacyNullToEmptyString] DOMString marginWidth;
-
-    [CheckSecurityForNode] Document getSVGDocument();
```
We should add a test to validate that this does not exist.

Ahmad-S792 replied:
@cdumez - Done and updated test to 'testharness' as well.

Ahmad-S792 committed:
Remove `svgDocument` interface from `HTMLFrameElement` to align with other browsers

https://bugs.webkit.org/show_bug.cgi?id=278069
rdar://133800070

Reviewed by Chris Dumez and Ryosuke Niwa.

This patch aligns WebKit with Web Specification [1]"

[1] https://html.spec.whatwg.org/multipage/embedded-content-other.html#svg-0

As per web specification, `getSVGDocument` should only be exposed on
'iframe', 'embed' and 'object'. This patch removes it from 'frame' element.

Additionally, we made our implementation to return null in 259905@main but
never synced our IDL interface for 'object' and 'embed', so this does and match
it with 'iframe'.

* Source/WebCore/html/HTMLEmbedElement.idl:
* Source/WebCore/html/HTMLFrameElement.idl:
* Source/WebCore/html/HTMLObjectElement.idl:
* LayoutTests/svg/custom/frame-getSVGDocument-expected.txt -> Renamed to interface-getSVGDocument-expected.txt:
* LayoutTests/svg/custom/frame-getSVGDocument.html -> Renamed to interface-getSVGDocument.html: Now updated to 'testharness' with all cases
* LayoutTests/platform/mac/svg/custom/frame-getSVGDocument-expected.png: Remove pixel output

Canonical link: https://commits.webkit.org/291498@main
